### ISSUE TITLE
[Feature Request] seamless Jupyter (IJulia) notebook editing in vscode

### ISSUE THREAD

**Issue by utotch** [julia-vscode/julia-vscode#874]
I want a seamless Jupyter editing environment in vscode as same as python Jupyter notebook support, which is released these days.

https://code.visualstudio.com/docs/python/jupyter-support

This is just a feature request.

**Comment by innerlee** [comment]
might be related: https://github.com/microsoft/vscode/pull/86632

**Comment by davidanthoff** [comment]
Yes, that looks very promising! Maybe we'll just get Jupyter notebook support in the VS Code base product :)

**Comment by davidanthoff** [comment]
Here is my current thinking on this: I believe that we won't add Jupyter notebook support in this extension here. Instead, I hope that a generic Jupyter notebook VS Extension will emerge that supports not just Python but also any other language.

I do plan to add support for a notebook editing interface for .jmd files once the notebook editor support in the core VS product ships. The whole experience will be modeled after the R Notebook model.

**Comment by ldsands** [comment]
It looks like [some are making progress](https://github.com/dotnet/interactive/pull/355) on this. (posting here in case it helps future development of this)

**Comment by davidanthoff** [comment]
Work on this is happening in https://github.com/julia-vscode/julia-vscode/pull/980.

**Comment by raghavmallampalli** [comment]
On Insider build, the notebook seems to rather seamlessly. I'm on the latest Insider build myself. I've tested the following features:
- Opening an existing ipynb notebook (created on Jupyter Lab)
- Creating a ipynb notebook
   - Opening this created notebook in Jupyter Lab
- Running cells
- Adding packages using `Pkg.add("Name")`
- Package support:
   - DifferentialEquations
   - Plots
   - DataFrames
- Plotting using GR backend
- Plotting using Plotly backend

Make sure you have:
- julia executable on path
- IJulia package added
- Jupyter extension for VS Code installed

Also keep in mind VS Code does not start the kernel until a cell is run (Ctrl+Alt+Enter or the Run button next to each cell)

**Comment by JontySinai** [comment]
In addition you need to make sure that the Julia kernel in the kernel spec list (run `jupyter kernelspec list`) and navigate to the path of the Julia kernel you want to use and ensure that the source path in kernel.json points to the Julia executable. This is the kernel VS Code will use as the Julia interpreter.

**Comment by storopoli** [comment]
Do you need to have the vscode insiders build nowadays? Or can it be done with the vscode "normal" build?

**Comment by davidanthoff** [comment]
Still VS Code Insider builds only, AFAIK.

**Comment by DonJayamanne** [comment]
@davidanthoff I noticed the Notebook branch hasn't been updated for a while.
I'm keen to get the Julia extension working with the Jupyter Extension (ipynb notebooks), with the Julia Kernel. I.e. Jupyter extension will just manage loading contents from ipynb and displaying & saving it, Jualia extension would own all of the execution & the like.

I was wondering if i could submit a PR (based on the original notebook branch) 
* That would support the Julia kernel (same code form Notebook branch)
* Get it working with Jupyter IPynb Files.

Happy to help, please let me know if this sounds good.
Oh yes, I'd be happy to setup the CI as well, ensure things work with insiders.
& I understand that none of them can be merged into the main branch just yet.

**Comment by davidanthoff** [comment]
@DonJayamanne Thanks for getting in touch, this is great timing! I was actually planning to reach out to you guys [1] to discuss how we can make this whole Julia kernel for Jupyter experience as smooth as possible! From my end that is really a high priority and I would be more than game that we try to figure this out jointly.

Might it actually make sense for the two of us (and anyone else from the core teams) to schedule a meeting to discuss a bit of a roadmap for this and how we can best do all of this? I've thought a fair bit about various aspects of this, but I'm also not super familiar at the moment with the plans for the Jupyter extension, so my sense is that a general exchange of where we are on the Jupyter and Julia side of things might make sense? My email is on my Github profile, do you just want to send me an email there to connect, and then we can schedule something?

[1] My understanding is that you are at MS and leading the Jupyter extension effort, right?

**Comment by DonJayamanne** [comment]
> My understanding is that you are at MS and leading the Jupyter extension effort, right?

Yes, but I wouldn't say I'm leading the Jupyter extension effort. I'm one of the engineers on the team.

> My email is on my Github profile, do you just want to send me an email there to connect, and then we can schedule something?

Will do.
